app/services/user_signup.rb

```diff
@@ -0,0 +1,47 @@
+class UserSignup
+  attr_reader :error_message
+
+  def initialize(user)
+    @user = user
+  end
+
+  def sign_up(stripe_token, invitation_token)
+    if @user.valid?
+      charge = StripeWrapper::Charge.create(
+        amount: 999,
+        source: stripe_token,
+        description: "MyFlix sign up charge for #{@user.email}"
+      )
+      if charge.successful?
+        @user.save
+        handle_invitation(invitation_token)
+        ApplicationMailer.send_welcome_email(@user).deliver
+        @status = :success
+        self
```

Comment:
Since all of your branches lead to `self`, you can refactor that to the end of the method from each branch of your code.